torch/csrc/distributed/c10d/init.cpp

```diff
@@ -700,7 +708,32 @@ They are used in specifying strategies for reduction collectives, e.g.,
           "wait",
           &::c10d::ProcessGroup::Work::wait,
           py::arg("timeout") = kNoTimeout,
-          py::call_guard<py::gil_scoped_release>());
+          py::call_guard<py::gil_scoped_release>())
+      .def(
+          "get_future",
+          [](::c10d::ProcessGroup::Work& work)
+              -> std::shared_ptr<jit::PythonFutureWrapper> {
+            return std::make_shared<jit::PythonFutureWrapper>(work.getFuture());
+          },
+          R"(
+            ``get_future`` retrieves a future associated with the completion of
+            ``c10d.ProcessGroup.work``. As an example, a future object can be
+            retrieved by ``fut = process_group.allreduce(tensors).get_future()``.
+
+            In the example above, if ``allreduce`` work was done on GPU using NCCL
+            backend, ``fut.wait()`` would return after synchronizing the appropriate
```

Comment:
> if ``allreduce`` work was done on GPU using NCCL

Does other backend support this API? If not, let's be explicit.

Reply:
It currently only supports NCCL. I revised it and mention API's warning inside this paragraph as well.